apps/yapms/src/lib/components/modals/importmodal/ImportModal.svelte

```diff
@@ -1,26 +1,50 @@
 <script lang="ts">
-	import ArrowUpTray from '$lib/icons/ArrowUpTray.svelte';
 	import { ImportModalStore } from '$lib/stores/Modals';
 	import { DOMPurifyConfig, importFromGeoJson, importFromShapefiles } from '$lib/utils/importMap';
 	import { ImportedSVGStore } from '$lib/stores/ImportedSVG';
 	import ExclamationCircle from '$lib/icons/ExclamationCircle.svelte';
 	import ModalBase from '../ModalBase.svelte';
-	import DocumentDuplicate from '$lib/icons/DocumentDuplicate.svelte';
 	import DOMPurify from 'dompurify';
-	import { page } from '$app/stores';
+	import { page } from '$app/state';
 	import { goto } from '$app/navigation';
 	import { exportImportAsSVG } from '$lib/utils/importMap';
-	import ImportOptions from './ImportOptions.svelte';
+	import GeoJsonOptions from './GeoJSONOptions.svelte';
+	import ProjectionOptions from './ProjectionOptions.svelte';
 
-	let geoJsonFiles: FileList;
-	let shapeFiles: FileList;
-	let svgFiles: FileList;
+	let files = $state<FileList | null | undefined>(undefined);
+	let loadError = $state<boolean>(false);
+	let loading = $state<boolean>(false);
 
-	let loadError = false;
-	let loading = false;
+	let fileType = $derived(
+		// set the file type to the mime type
+		// if a mime type isn't detected, then use the file extension
+		Array.from(files || []).reduce((accu, curr) => {
+			let type: string | undefined = undefined;
 
-	const multipleFilesTooltip =
-		'Select multiple files to upload and they will be automatically merged.';
+			if (curr.type === 'application/json') {
+				type = 'application/geo+json';
+			} else if (curr.type !== '') {
+				type = curr.type;
+			}
+
+			if (type === undefined) {
+				const extension = curr.name.split('.').at(-1) ?? '';
+				type = extension;
+			}
+
+			if (accu === '' || accu === type) {
+				return type;
+			}
+
+			return 'multiple';
+		}, '')
+	);
+
+	let fileTypeIsInvalid = $derived(
+		files &&
+			files.length > 0 &&
+			['application/geo+json', 'shp', 'image/svg+xml'].includes(fileType) === false
```

Comment:
This does not handle the .geojson file ending correctly. (I guess the MIME type is not detected correctly)

Reply:
Also is not set to true when files === undefined or when files.length == 0 as appears to be intended here.

Reply:
My though is fixes here will properly disable the load button when you first open the modal and when files.length is 0.

Additionally, it is possible under the current implementation to upload a mix of .geojson and .json files and it will work. I would appreciate if we could maintain that.